src/libstore/tests/derived-path.cc

```diff
@@ -0,0 +1,52 @@
+#include <regex>
+
+#include <nlohmann/json.hpp>
+#include <gtest/gtest.h>
+#include <rapidcheck/gtest.h>
+
+#include "tests/derived-path.hh"
+#include "tests/libstore.hh"
+
+namespace rc {
+using namespace nix;
+
+Gen<DerivedPath> Arbitrary<DerivedPath>::arbitrary()
+{
+    switch (*gen::inRange<uint8_t>(0, 1)) {
+    case 0:
+        return gen::just((DerivedPath) DerivedPath::Opaque {
+            .path = *gen::arbitrary<StorePath>(),
+        });
+    default:
+        return gen::just((DerivedPath) DerivedPath::Built {
+            .drvPath = *gen::arbitrary<StorePath>(),
+            .outputs = *gen::arbitrary<OutputsSpec>(),
+        });
+    }
```

Comment:
Maybe these should be granular instances, as each constructor is also a type on its own that may be useful at times. It might also just be nicer to read and write that way, following the declarations more closely.

Reply:
Done